lib/dbservice_web/controllers/student_controller.ex

```diff
@@ -158,4 +163,72 @@ defmodule DbserviceWeb.StudentController do
       |> render("show.json", student: student)
     end
   end
+
+  def dropout(conn, %{"student_id" => student_id}) do
+    student = Users.get_student_by_student_id(student_id)
+
+    # Check if the student's status is already 'dropout'
+    if student.status == "dropout" do
+      conn
+      |> put_status(:bad_request)
+      |> json(%{error: "Student is already marked as dropout"})
+    else
+      user_id = student.user_id
+      current_time = DateTime.utc_now()
+
+      # Fetch status and group details in a single query
+      {group_id, group_type} =
+        from(s in Status,
+          join: g in Group,
+          on: g.child_id == s.id and g.type == "status",
+          where: s.title == "dropout",
+          select: {g.id, g.type}
+        )
+        |> Repo.one()
+
+      # Update current enrollment records for the user
+      current_enrollment =
+        from(e in EnrollmentRecord,
+          where: e.user_id == ^user_id and e.is_current == true
+        )
+        |> Repo.one()
+
+      if current_enrollment do
+        academic_year = current_enrollment.academic_year
+        grade_id = current_enrollment.grade_id
+
+        EnrollmentRecords.update_enrollment_record(current_enrollment, %{
+          is_current: false,
+          end_date: current_time
+        })
+
+        # Create a new enrollment record with the fetched group_id
+        new_enrollment_attrs = %{
+          user_id: user_id,
+          is_current: true,
+          start_date: current_time,
+          group_id: group_id,
```

Comment:
what would this group_id refer to? where group_type is status?

Reply:
`group_id` is the ID (PK) of `group`. Each record have `type` and `child_id` in it. That `child_id` will refer PK of `status` table